metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py

```diff
@@ -760,19 +772,69 @@ def create_lineage(
 
 
 class DatabrickDataPlatformTableCreator(AbstractDataPlatformTableCreator):
+    def form_qualified_table_name(
+        self,
+        value_dict: Dict[Any, Any],
+        catalog_name: str,
+        data_platform_pair: DataPlatformPair,
+        server: str,
+    ) -> str:
+        # database and catalog names are same in M-Query
+        db_name: str = (
+            catalog_name if "Database" not in value_dict else value_dict["Database"]
+        )
+
+        schema_name: str = value_dict["Schema"]
+
+        table_name: str = value_dict["Table"]
+
+        platform_detail: PlatformDetail = (
+            self.platform_instance_resolver.get_platform_instance(
+                PowerBIPlatformDetail(
+                    data_platform_pair=data_platform_pair,
+                    data_platform_server=server,
+                )
+            )
+        )
+
+        metastore: Optional[str] = None
+
+        if isinstance(platform_detail, DataBricksPlatformDetail):
+            metastore = cast(DataBricksPlatformDetail, platform_detail).metastore
```

Comment:
the cast is not required

Reply:
also move this if statement to after the initial creation of `qualified_table_name`